File: frontend/src/models/RecipeCategory.ts
import { Recipe } from "./Recipe";

export interface RecipeCategory {
	id: number;
	name: string;
	recipes?: Recipe[];
}

export const RECIPE_CATEGORY_CONSTANTS = {
	CATEGORIES: [
		{ id: 1, name: "Food" },
		{ id: 2, name: "Dessert" }
	]
};

File: Backend/Src/Models/RecipeCategory.cs
using System.ComponentModel.DataAnnotations;

namespace GourmetGateway.Models;

public partial class RecipeCategory
{
	public int Id { get; set; }

	[MinLength(3), MaxLength(15)]
	public required string Name { get; set; }

	public virtual ICollection<Recipe> Recipes { get; set; } = [];
}